File: src/main/java/cn/don9cn/blog/action/bussiness/articleclassify/ArticleClassifyAction.java
package cn.don9cn.blog.action.bussiness.articleclassify;

import cn.don9cn.blog.action.BaseAction;
import cn.don9cn.blog.model.bussiness.articleclassify.ArticleClassify;
import cn.don9cn.blog.plugins.daohelper.core.PageResult;
import cn.don9cn.blog.plugins.operaresult.core.OperaResult;
import cn.don9cn.blog.service.bussiness.articleclassify.interf.ArticleClassifyService;
import org.apache.log4j.Logger;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;


/** 
 * @Title: ArticleClassifyAction.java
 * @Package cn.don9.action.article
 * @Description: ArticleClassify模块action
 * @author liuxindong
 * @createDate 2017-08-11 16:31:11
 */
@RestController
@RequestMapping(value = "/bussiness/articleClassify")
public class ArticleClassifyAction extends BaseAction<ArticleClassify> {
	
    private static Logger logger = Logger.getLogger(ArticleClassifyAction.class);

	@Autowired
	private ArticleClassifyService articleClassifyService;


	@Override
	@PostMapping
	protected Object baseInsert(ArticleClassify articleClassify) {
		return articleClassifyService.baseInsert(articleClassify);
	}

	@Override
	protected Object baseInsertBatch(List<ArticleClassify> list) {
		return null;
	}

	@Override
	@PutMapping
	protected Object baseUpdate(ArticleClassify articleClassify) {
		return articleClassifyService.baseUpdate(articleClassify);
	}

	@Override
	protected Object baseRemove(String code) {
		return null;
	}

	@Override
	@DeleteMapping
	protected Object baseRemoveBatch(String codes) {
		return articleClassifyService.baseDeleteBatch(codes);
	}

	@Override
	@GetMapping
	protected Object baseFindById(String code) {
		return articleClassifyService.baseFindById(code);
	}

	@Override
	@GetMapping("/all")
	protected Object baseFindAll() {
		return articleClassifyService.baseFindAll();
	}

	@Override
	@GetMapping("/list")
	protected Object baseFindListByParams(ArticleClassify articleClassify) {
		return articleClassifyService.baseFindListByParams(articleClassify);
	}

	@Override
	@GetMapping("/page")
	protected Object baseFindByPage(int page,int limit,String startTime,String endTime,String orderBy,ArticleClassify articleClassify) {
		return articleClassifyService.baseFindByPage(new PageResult<>(page,limit,startTime,endTime,orderBy,articleClassify));
	}

	/**
	 * 获取分类下拉列表
	 * @return
	 */
	@GetMapping("/selectOptions")
	public Object doGetSelectOptions() {
		return articleClassifyService.doGetSelectOptions();
	}

	/**
	 * 获取分类树
	 * @return
	 */
	@GetMapping(path = {"/tree","/tree/public"})
	public Object getTree() {
		return articleClassifyService.getTree();
	}
}

File: src/main/java/cn/don9cn/blog/action/bussiness/article/ArticleClassifyAction.kt
package cn.don9cn.blog.action.bussiness.article

import cn.booklish.mongodsl.core.PageResult
import cn.don9cn.blog.action.BaseAction
import cn.don9cn.blog.model.bussiness.articleclassify.ArticleClassify
import cn.don9cn.blog.service.bussiness.ArticleClassifyService
import cn.don9cn.blog.service.bussiness.ArticleService
import cn.don9cn.blog.support.action.ActionMsg
import cn.don9cn.blog.support.action.ActionMsgHandler
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = "/bussiness/articleClassify")
open class ArticleClassifyAction : BaseAction<ArticleClassify>() {

    @Autowired
    private var articleClassifyService: ArticleClassifyService? = null



    @PostMapping
    override fun baseInsert(articleClassify: ArticleClassify): ActionMsg {
        return ActionMsgHandler.insert(articleClassifyService!!.baseInsert(articleClassify))
    }

    override fun baseInsertBatch(list: List<ArticleClassify>): ActionMsg {
        return ActionMsgHandler.insert(articleClassifyService!!.baseInsertBatch(list))
    }

    @PutMapping
    override fun baseUpdate(articleClassify: ArticleClassify): ActionMsg {
        return ActionMsgHandler.update(articleClassifyService!!.baseUpdate(articleClassify))
    }

    override fun baseRemove(code: String): ActionMsg {
        return ActionMsgHandler.delete(articleClassifyService!!.baseDeleteById(code))
    }

    @DeleteMapping
    override fun baseRemoveBatch(codes: String): ActionMsg {
        return ActionMsgHandler.delete(articleClassifyService!!.baseDeleteBatch(codes))
    }

    @GetMapping
    override fun baseFindById(code: String): ActionMsg {
        return ActionMsgHandler.find(articleClassifyService!!.baseFindById(code))
    }

    @GetMapping("/all")
    override fun baseFindAll(): ActionMsg {
        return ActionMsgHandler.find(articleClassifyService!!.baseFindAll())
    }

    @GetMapping("/list")
    override fun baseFindListByParams(articleClassify: ArticleClassify): ActionMsg {
        return ActionMsgHandler.find(articleClassifyService!!.baseFindListByParams(articleClassify))
    }

    @GetMapping("/page")
    override fun baseFindByPage(page: Int, limit: Int, startTime: String?, endTime: String?, orderBy: String?, articleClassify: ArticleClassify): ActionMsg {
        return ActionMsgHandler.find(articleClassifyService!!.baseFindByPage(PageResult(page, limit, startTime, endTime, orderBy, articleClassify)))
    }

    /**
     * 获取分类树
     */
    @GetMapping(path = ["/tree", "/tree/public"])
    open fun getTree():ActionMsg{
        return ActionMsgHandler.find(articleClassifyService!!.getTree())
    }

    /**
     * 获取分类下拉列表
     */
    @GetMapping("/selectOptions")
    open fun doGetSelectOptions(): ActionMsg {
        return ActionMsgHandler.find(articleClassifyService!!.doGetSelectOptions())
    }

}